pkg/downloader/downloader.go

```diff
@@ -600,10 +598,8 @@ func downloadHTTP(ctx context.Context, localPath, lastModified, contentType, url
 		return fmt.Errorf("downloadHTTP: got empty localPath")
 	}
 	logrus.Debugf("downloading %q into %q", url, localPath)
-	localPathTmp := localPath + ".tmp"
-	if err := os.RemoveAll(localPathTmp); err != nil {
-		return err
-	}
+
+	localPathTmp := perProcessTempfile(localPath)
```

Comment:
Can we `defer os.RemoveAll(localPathTmp)` to minimize the possibility of leaking the tmp file?

Reply:
perProcessTempfile is a new code though